src/scripts/main.js

```diff
@@ -1,3 +1,224 @@
 'use strict';
 
-// write code here
+const DESC = 'desc';
+const ASC = 'asc';
+const HEADERS = ['name', 'position', 'office', 'age', 'salary'];
+const OFFICES = [
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+];
+
+const itemCollection = document.querySelectorAll('tbody tr');
+const tableHeader = document.querySelector('thead tr');
+const tableBody = document.querySelector('tbody');
+
+const items = [...itemCollection].map((item) => {
+  const collection = item.querySelectorAll('td');
+
+  return HEADERS.reduce(
+    (prev, current, index) => ({
+      ...prev,
+      [current]: collection[index].innerText,
+    }),
+    {},
+  );
+});
+
+function rebuildList(data) {
+  tableBody.innerHTML = '';
+
+  for (const item of data) {
+    const row = document.createElement('tr');
+
+    for (const key in item) {
+      const rowData = document.createElement('td');
+
+      rowData.innerText = item[key];
+      row.appendChild(rowData);
+    }
+
+    tableBody.appendChild(row);
+  }
+}
+
+tableHeader.addEventListener('click', (e) => {
+  const title = e.target.closest('th');
+
+  if (!title) {
+    return;
+  }
+
+  const targetValue = title.innerText.toLowerCase();
+
+  if (!title.hasAttribute('data-order') || title.dataset.order === DESC) {
+    title.setAttribute('data-order', ASC);
+  } else {
+    title.setAttribute('data-order', DESC);
+  }
+
+  items.sort((a, b) => {
+    const order = title.dataset.order;
+
+    if (targetValue === 'salary') {
+      const aSalary = a[targetValue].slice(1).split(',').join('');
+      const bSalary = b[targetValue].slice(1).split(',').join('');
+
+      return order === ASC ? aSalary - bSalary : bSalary - aSalary;
+    }
+
+    if (targetValue === 'age') {
+      return order === ASC
+        ? a[targetValue] - b[targetValue]
+        : b[targetValue] - a[targetValue];
+    }
+
+    return order === ASC
+      ? a[targetValue].localeCompare(b[targetValue])
+      : b[targetValue].localeCompare(a[targetValue]);
+  });
+
+  rebuildList(items);
+});
```

Comment:
The sorting logic inside the event listener can be moved to a separate function. This will make the event listener code cleaner and easier to understand. The function can be named something like 'sortItems'.